shiny/render/_dataframe.py

```diff
@@ -0,0 +1,320 @@
+from __future__ import annotations
+
+import abc
+import json
+import typing
+from typing import Any, Awaitable, Callable, Literal, Optional, Union, cast, overload
+
+from .. import _utils
+from .._typing_extensions import Protocol, runtime_checkable
+from . import RenderFunction, RenderFunctionAsync
+
+
+class AbstractTabularData(abc.ABC):
+    @abc.abstractmethod
+    def to_payload(self) -> object:
+        ...
+
+
+class DataGrid(AbstractTabularData):
+    def __init__(
+        self,
+        data: object,
+        *,
+        width: Union[str, float, None] = "fit-content",
+        height: Union[str, float, None] = "500px",
+        summary: Union[bool, str] = True,
+        row_selection_mode: Union[
+            Literal["none"], Literal["single"], Literal["multi-toggle"]
+        ] = "none",
```

Comment:
This can be written more concisely:

```suggestion
        row_selection_mode: Literal["none", "single", "multi-toggle"] = "none",
```

Reply:
Whoa, how did I not know that?